stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+                flagReportingVerb = False
+                for token in textTokens:
+                    if token.pos.startswith("VB") and token.lemma.lower() in self.reportingVerbs:
+                        flagReportingVerb = True
+                        break
+                if flagReportingVerb == False:
+                    continue
+
+                mention = currentQuote.mention
+                mention_start = currentQuote.mention_start
+                mention_end = currentQuote.mention_end
+
+                # Current <- Next
+                if mention is None:
+                    if nextQuote.mention is not None:
+                        self.quotes[i].mention = nextQuote.mention
+                        self.quotes[i].mention_start = nextQuote.mention_start
+                        self.quotes[i].mention_end = nextQuote.mention_end
+                        mention = nextQuote.mention
+                        mention_start = nextQuote.mention_start
+                        mention_end = nextQuote.mention_end
+                    continue
+                candidateQuoteIds = []
+                for qObj in self.quoteParaDict[currentQuote.paragraphId]:
+                    if qObj.quoteListIdx > currentQuote.quoteListIdx:
+                        candidateQuoteIds.append(qObj.quoteListIdx)
+
+                for qIdx in candidateQuoteIds:
+                    # Current -> Next
+                    if mention is not None:
+                        if self.quotes[qIdx].mention is None:
+                            self.quotes[qIdx].mention = mention
+                            self.quotes[qIdx].mention_start = mention_start
+                            self.quotes[qIdx].mention_end = mention_end
+                            continue
+
+    def createConversationCsv(self, basepath, filename):
+        conversationQuotes = []
+        for convo in self.conversations:
+            quotesMerged = ""
+            for each in convo:
+                quotesMerged += f"{each.mention} : " + each.quote + "\n"
+            conversationQuotes.append({'quotes': quotesMerged})
+        df = pd.DataFrame(conversationQuotes)
+        df.to_csv(basepath + filename + '.csv')
+
+    def getConversations(self):
+        conversations = []
+        currentConversation = []
+
+        for i, quoteObj in enumerate(self.quotes):
+            if len(currentConversation) == 0:
+                currentConversation.append(quoteObj)
+            else:
+                previousSentenceId = int(currentConversation[-1].endSentId)
+                if int(quoteObj.startSentId) - previousSentenceId <= 3:
+                    currentConversation.append(quoteObj)
+                else:
+                    conversations.append(currentConversation)
+                    currentConversation = [quoteObj]
+
+        self.conversations = conversations
+
+    def searchUpwards(self, conversationIdx, currentIdx):
```

Comment:
searchUpwards and searchDownwards have duplicate code that should be merged logically. There should be one single function that takes in a boolean parameter to either search upward or downward